src/main/java/frc/robot/subsystems/manipulator/ManipulatorIOSparkMax.java

```diff
@@ -0,0 +1,63 @@
+package frc.robot.subsystems.manipulator;
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMaxLowLevel;
+import com.revrobotics.RelativeEncoder;
+import com.revrobotics.SparkMaxPIDController;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.RobotController;
+import frc.robot.subsystems.flywheel.FlywheelIO;
+
+public class ManipulatorIOSparkMax implements ManipulatorIO {
+    public static final double GEAR_RATIO = 1;
+    public static final int deviceId = 6; //TODO fill this in
+    private final CANSparkMax motor;
+    private final RelativeEncoder encoder;
+    private final SparkMaxPIDController pid;
+
+    public ManipulatorIOSparkMax(){
+        motor = new CANSparkMax(deviceId, CANSparkMaxLowLevel.MotorType.kBrushless);
+
+        encoder = motor.getEncoder();
+        pid = motor.getPIDController();
+
+        motor.restoreFactoryDefaults();
+
+        motor.setInverted(false);
+
+        motor.enableVoltageCompensation(12.0);
+        motor.setSmartCurrentLimit(30);
+
+        motor.burnFlash();
+    }
+
+    @Override
+    public void updateInputs(ManipulatorIOInputs inputs) {
+        inputs.positionRad = Units.rotationsToRadians(encoder.getPosition() / GEAR_RATIO);
+        inputs.velocityRadPerSec = Units.rotationsPerMinuteToRadiansPerSecond(
+                encoder.getVelocity() / GEAR_RATIO);
+        inputs.appliedVolts = motor.getAppliedOutput() * RobotController.getBatteryVoltage();
+        inputs.currentAmps = new double[] {motor.getOutputCurrent()};
+    }
+
+    @Override
+    public void setVelocity(double velocityRadPerSec, double ffVolts) {
+        pid.setReference(
+                Units.radiansPerSecondToRotationsPerMinute(velocityRadPerSec)
+                        * GEAR_RATIO,
+                CANSparkMax.ControlType.kVelocity, 0, ffVolts, SparkMaxPIDController.ArbFFUnits.kVoltage);
+    }
```

Comment:
```suggestion
```

we dont need pid (telling a motor to set it's velocity is telling it to use a pid loop)